langchain/vectorstores/azuresearch.py

```diff
@@ -171,21 +181,74 @@ def __init__(
         search_type: str = "hybrid",
         semantic_configuration_name: Optional[str] = None,
         semantic_query_language: str = "en-us",
+        fields: list[SearchField] = None,
+        vector_search: VectorSearch = None,
+        semantic_settings: SemanticSettings = None,
+        scoring_profiles: Optional[list[ScoringProfile]] = None,
+        default_scoring_profile: Optional[str] = None,
         **kwargs: Any,
     ):
+        from azure.search.documents.indexes.models import (
+            PrioritizedFields,
+            SearchableField,
+            SearchField,
+            SearchFieldDataType,
+            SearchIndex,
+            SemanticConfiguration,
+            SemanticField,
+            SemanticSettings,
+            SimpleField,
+            VectorSearch,
+            VectorSearchAlgorithmConfiguration,
+        )
         """Initialize with necessary components."""
         # Initialize base class
         self.embedding_function = embedding_function
+        default_fields = [
+            SimpleField(
+                name=FIELDS_ID,
+                type=SearchFieldDataType.String,
+                key=True,
+                filterable=True,
+            ),
+            SearchableField(
+                name=FIELDS_CONTENT,
+                type=SearchFieldDataType.String,
+                searchable=True,
+                retrievable=True,
+            ),
+            SearchField(
+                name=FIELDS_CONTENT_VECTOR,
+                type=SearchFieldDataType.Collection(SearchFieldDataType.Single),
+                searchable=True,
+                dimensions=len(embedding_function("Text")),
+                vector_search_configuration="default",
+            ),
+            SearchableField(
+                name=FIELDS_METADATA,
+                type=SearchFieldDataType.String,
+                searchable=True,
```

Comment:
Do you really want this to be searchable? It has a bunch of JSON in it, dosen't it? Does it make sense to include in search queries?

Reply:
Kept as it was in the original implementation. Happy to chat through this one :)